protostar/scripting/deployments/declare.py

```diff
@@ -0,0 +1,18 @@
+from dataclasses import dataclass
+from typing import Optional
+
+from protostar.scripting.typing import Fee
+
+
+@dataclass
+class DeclaredContract:
+    class_hash: int
+
+
+def declare(
+    contract: str,
+    *,
+    wait_for_acceptance: Optional[bool] = None,
```

Comment:
I wonder if we shouldn't make `wait_for_acceptance` be `True` by default, and non-optional, i.e. use `wait_for_acceptance: bool = True` everywhere

Reply:
Maybe `wait_for_acceptance` should be a generic function.

Reply:
> Maybe `wait_for_acceptance` should be a generic function.

I don't understand, could you elaborate please?

Reply:
```py
declare_tx = declare("main")
awaited_declare_tx = wait_for_acceptance(declare_tx)
assert isinstance(awaited_declare_tx, DeclareTx)
```

Reply:
Isn't this too noisy?

Also this sounds like futures, maybe we should reuse that ones?

Reply:
FYI I can allow top-level awaits in these scripts